From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Thu, 26 Feb 2026 12:31:04 +0100
Subject: remote-desktop: Correctly handle missing secret collection

The default collection is missing on a fresh user login with auto-login.
In that case, NULL is a valid return value for secret_collection_for_alias
and does not necessarily indicate an error, so error will not have been set.

The function signature for secret_collection_for_alias_sync correctly annotates
the return value as (nullable).

Fixes: 5940af5b20 ("remote-desktop, desktop-sharing: Handle when keyring is locked")

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3328
---
 panels/system/remote-desktop/cc-desktop-sharing-page.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/panels/system/remote-desktop/cc-desktop-sharing-page.c b/panels/system/remote-desktop/cc-desktop-sharing-page.c
index e10c56f..686a00b 100644
--- a/panels/system/remote-desktop/cc-desktop-sharing-page.c
+++ b/panels/system/remote-desktop/cc-desktop-sharing-page.c
@@ -461,7 +461,9 @@ unlock_gnome_keyring (CcDesktopSharingPage *self)
                                                  NULL, &error);
   if (!collection)
     {
-      g_debug ("Can't load secret collection: %s", error->message);
+      const char *msg = error ? error->message : "No such collection";
+
+      g_debug ("Can't load secret collection: %s", msg);
       return;
     }
 
